onnxruntime/core/providers/cuda/cuda_execution_provider.h

```diff
@@ -92,6 +94,16 @@ class CUDAExecutionProvider : public IExecutionProvider {
     return IAllocator::MakeUniquePtr<T>(GetAllocator(info_.device_id, OrtMemTypeDefault), count_or_bytes, true);
   }
 
+  template <typename T>
+  IAllocatorUniquePtr<T> AllocateBufferOnCPUPinned(size_t count_or_bytes) const {
```

Comment:
This function is moved from `CudaKernel::AllocateBufferOnCPUPinned` because the allocation and release of a specific type of memory should be handled by the same class. Previously, `CudaKernel` allocates but `CUDAExecutionProvider` frees pinned CPU memory.